src/router.js

```diff
@@ -11,80 +11,109 @@
  */
 import routington from 'routington';
 import { RestError } from './rest-error.js';
+import { ContextHelper } from './context.js';
 
 /**
- * Function for handling a routes inside Frankin / Content Lake services
+ * @typedef SecurityConfig
+ * @property {boolean} authRequired;
+ * @property {Array<string>} [allowedRoles]
+ * @property {Array<string>} [allowedPermissions]
+ */
+
+/**
+ * Function for handling a routes inside Franklin / Content Lake services
  * @callback Handler
- * @param {Request} req the request
- * @param {UniversalContext} context the context of the request
  * @param {Record<string,string>} params the parameters parsed from the request
+ * @param {{request:Request,context:import('./context.js').UniversalishContext}} info the
+ *   additional request information
  * @returns {Promise<Response>} the response from the request
  */
 
 export class Router {
   methods = {};
 
+  /**
+   * @type {import('./security.js').Security | undefined}
+   */
+  #security;
+
   /**
    *
    * @param {string} method
    * @param {string} path
    * @param {Handler} handler
+   * @param {SecurityConfig} [securityConfig]
```

Comment:
is this new param required?

Reply:
Nope, if not provided no security will be enforced (ex: Oauth callbacks, updates, etc)